Kentor.AuthServices/WebSSO/Saml2ArtifactBinding.cs

```diff
@@ -90,22 +90,19 @@ public override UnbindResult Unbind(HttpRequestData request, IOptions options)
             var arsIndex = (binaryArtifact[2] << 8) | binaryArtifact[3];
             var arsUri = idp.ArtifactResolutionServiceUrls[arsIndex];
 
-            var payload = new Saml2ArtifactResolve()
+            var payload = new Saml2ArtifactResolve
             {
                 Artifact = artifact,
                 Issuer = options.SPOptions.EntityId
             }.ToXml();
 
-            if (options.SPOptions.SigningServiceCertificate != null)
-            {
-                var xmlDoc = XmlHelpers.XmlDocumentFromString(payload);
-                xmlDoc.Sign(options.SPOptions.SigningServiceCertificate, true);
-                payload = xmlDoc.OuterXml;
-            }
```

Comment:
What is the significance of moving the payload signing from this location and into `SendSoapRequest`?